### DISCUSSION TITLE
Dynamic child components are broken

### DISCUSSION THREAD

**Discussion by CovertError** [livewire/livewire#7866] [7. Bugs (archived)]
### Livewire version

v3.4.4

### Laravel version

v10.28.0

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

check this repo just click on the next button it doesn't go to the next component 

### Repository or Wirebox URL

https://github.com/CovertError/LivewireTest

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by danie-ramdhani** [comment]
this not bug. you just need to set the unique key:

```blade
{{--  counter.blade.php --}}
<div>
    <livewire:dynamic-component :is="$current" :key="$current" />
    <button wire:click="next">Next</button>
</div>
```

here's the documentation: https://livewire.laravel.com/docs/components#adding-wirekey-to-foreach-loops

**Reply by joshhanley** [reply]
@CovertError yeah @danie-ramdhani is correct, you need keys on any components in a loop or dynamic components.

**Reply by CovertError** [reply]
> @CovertError yeah @danie-ramdhani is correct, you need keys on any components in a loop or dynamic components.

It's not mentioned in the docs

**Reply by CovertError** [reply]
> > @CovertError yeah @danie-ramdhani is correct, you need keys on any components in a loop or dynamic components.
> 
> It's not mentioned in the docs 

https://livewire.laravel.com/docs/nesting#dynamic-child-components

**Reply by joshhanley** [reply]
@CovertError ah yeah, probably just an oversight. PRs always welcome! 🙂

**Reply by legecha** [reply]
I'd be happy to submit a PR for this (just been through the process of figuring out the `key` part was missing from the documentation before finding this issue) - but is the new 3.0 site/docs under version control and accepting PRs?

**Reply by joshhanley** [reply]
@legecha Thank! Yep docs are in this repo. PRs are welcome!

**Reply by legecha** [reply]
@joshhanley Cheers. I've created PR #8461
